API MFA is always on, and cannot be disabled

NightJar opened the pull request:
MFA once enabled (i.e. installed) should not be able to be universally
disabled by an administrator. They can alter the "required-ness" of MFA,
i.e. whether or not users must register a second factor method or not,
but only users can add and remove registered methods from their account.
Once set up, MFA is always protecting that that account, whether
required by the site or not.

Closes #144 

![image](https://user-images.githubusercontent.com/778003/59318341-cf69e300-8d1a-11e9-8e02-de036fafe076.png)

newleeland commented:
Hey @NightJar looking good. I've got a just text changes to make this clearer.

**Text changes**
![59318341-cf69e300-8d1a-11e9-8e02-de036fafe076-2](https://user-images.githubusercontent.com/8085295/59320905-422b8c00-8d24-11e9-8287-759cbedd2912.png)

Also, is there a reason why the radio labels are not aligned with radio?

robbieaverill commented on client/src/legacy/SiteConfig.js:
```diff
@@ -1,45 +1,14 @@
-/* global window */
-const hiddenClass = 'mfa-settings--hidden';
```
Is there some CSS to remove here too?

NightJar replied:
I believe I've nabbed it already (in `SiteConfigExtension.php`), but can search to be sure that's all of it :)

robbieaverill replied:
See SiteConfig.scss in the legacy folder

NightJar replied:
Ah, you meant the scss rule :joy:

robbieaverill commented on src/Extension/SiteConfigExtension.php:
```diff
@@ -43,30 +41,27 @@ public function updateCMSFields(FieldList $fields)
         );
+        $mfaGraceEnd->setDescription(_t(
+            __CLASS__ . '.MFA_GRACE_DESCRIPTION',
+            'MFA setup will be optional during this grace period, and reminders to set up will appear at log in.'
```
Possible improvement: "and reminders to register will appear at log in."

NightJar replied:
I was thinking that as I typed it, but was also aware that changes were inbound from PUX, so didn't think about it too much. The more one thinks about it the more and more "wordy" it has the propensity to become :upside_down_face:

NightJar commented:
> Hey @NightJar looking good. I've got a just text changes to make this clearer.

Neat, thanks!

> **Text changes**
> ![59318341-cf69e300-8d1a-11e9-8e02-de036fafe076-2](https://user-images.githubusercontent.com/8085295/59320905-422b8c00-8d24-11e9-8287-759cbedd2912.png)

One suggestion about the first suggestion (changing "everyone" to "CMS acces") - currently the latter is entirely untrue. I'd like to think that if we were to have time to get around to the issue that deals with different user permissions as to who is required to have MFA and who isn't, then we'd make this change as part of that.

> Also, is there a reason why the radio labels are not aligned with radio?

Nope, hahaha just hadn't noticed! I think this might be default (i.e. `admin`) behaviour for radio buttons. I'll see if I can find another set to check against.
If it is, and no one else can replicate, then I'd be inclined to putting it down as "this is neither Windows nor Mac" and differences in font rendering engines.

NightJar commented:
OK, I've updated the text, save the one I pointed out above. (see #179 )

Also removed the now redundant rule from the css bundle :)

I've double checked that the text on the radio buttons is not unique to the MFA switch, and am marking that down as Linux font rendering related (see other radio buttons have the same issue in the expanded screenshot below).

![image](https://user-images.githubusercontent.com/778003/59323125-78213e00-8d2d-11e9-96b6-01c627a01876.png)

robbieaverill commented on src/Extension/SiteConfigExtension.php:
```diff
@@ -43,30 +41,27 @@ public function updateCMSFields(FieldList $fields)
         $mfaGraceEnd = DateField::create(
             'MFAGracePeriodExpires',
-            'Optional: grace period end date when MFA is enforced'
+            'MFA will be required from (optional)'
```
Needs translation

robbieaverill commented on src/Extension/SiteConfigExtension.php:
```diff
@@ -91,7 +86,7 @@ protected function getHelpLink()
             '<a class="d-block mfa-settings__help-link" target="blank" rel="noopener" href="%s">%s</a>',
             $link,
-            _t(__CLASS__ . '.FIND_OUT_MORE', 'Find out more')
+            _t(__CLASS__ . '.FIND_OUT_MORE', 'Learn about MFA')
```
This has a different purpose to the translation key now, maybe rename the key as well
